### ISSUE TITLE
"recurse" feature in Find module is not working for file_type = directory option  

### ISSUE THREAD

**Issue by Ritesh-Sharma** [ansible/ansible#19870]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME
File Module 

##### ANSIBLE VERSION
ansible 2.2.0.0
```

```
##### CONFIGURATION
Default

##### OS / ENVIRONMENT
 Ubuntu 14.04.5 LTS

##### SUMMARY

Hi Team , 

I have some shell scripts which are dumping machine stats on remote nodes (of those remote machines only ) in new directory . this new directory has been created every time (on static path ) the shell scripts are being started.  

I am expecting directory name as a result , whose contained files are being updated at run time (while running playbook ) .

I am using File module but it seems recurse="yes"  is not working with file_type="directory"

```yaml

---
- hosts: wallet
  tasks:
    - name: Verifying existence of stat directory
      file: path=/tmp/ansibleTest  state=directory

    - name: Identify current logging directory 
      find: paths="/tmp/ansibleTest/perf_output" file_type="directory" age="-2m" recurse="yes" age_stamp="mtime"
      register: result

    - debug: var=result

    - set_fact: path="{{ item }}"
      with_items: "{{ result.files[0].path }}"

    - debug: var=path


```

##### EXPECTED RESULTS
so, basically i am trying to fetch directory name whose containing "Files" are having mtime <= 2 minutes   

##### ACTUAL RESULTS

```
I  am getting a directory name as a output whose mtime <= 2 minutes  .

```

Kindly Help ?

**Comment by bcoca** [comment]
Possible Misunderstanding
=========================

Hi!

Thanks very much for your submission to Ansible.  It sincerely means a lot to us.

We believe the ticket you have filed is being somewhat misunderstood, as one thing works a little differently than stated.

When using `file_type="directory"`, ONLY directories get examined, not their contents. So in this case the expectation is not matching the documentation nor functionality. To get what you want you need to examine 'files', not directories, then use the `dirname` filter to get your list.

In the future, this might be a topic more well suited for the user list, which you can also post here if you'd like some more help with the above.
   
   * https://groups.google.com/forum/#!forum/ansible-project

Thank you once again for this and your interest in Ansible!

**Comment by Ritesh-Sharma** [comment]
Ok Thanks , bcoca .

**Comment by bcoca** [comment]
As an example:
```
- hosts: wallet
  tasks:
    - name: Verifying existence of stat directory
      file: path=/tmp/ansibleTest  state=directory

    - name: Identify current logging directory 
      find: paths="/tmp/ansibleTest/perf_output" file_type="file" age="-2m" recurse="yes" age_stamp="mtime"
      register: result

    - debug: var=result

    - set_fact: path="{{ result.files|map(attribute='path')|map('dirname')|unique }}"

    - debug: var=path
```
